content/en/blog/_posts/2023-12-14-disabling-in-tree-cloud-provider-goes-beta.md

```diff
@@ -0,0 +1,210 @@
+
+In 2018, the [Kubernetes community agreed to form the Cloud Provider Special
+Interest Group (SIG)][oldblog], with a mission to externalize all cloud provider
+integrations and remove all the existing in-tree cloud provider integrations.
+In January 2019, the Kubernetes community approved the initial draft of
+[KEP-2395: Removing In-Tree Cloud Provider Code][kep2395]. This KEP defines a
+process by which we can remove cloud provider specific code from the core
+Kubernetes source tree. From the KEP:
+
+> Motiviation behind this effort is to allow cloud providers to develop and
+> make releases independent from the core Kubernetes release cycle. The
+> de-coupling of cloud provider code allows for separation of concern between
+> "Kubernetes core" and the cloud providers within the ecosystem. In addition,
+> this ensures all cloud providers in the ecosystem are integrating with
+> Kubernetes in a consistent and extendable way.
+
+After many years of development and collaboration across many contributors,
+the default behavior for legacy cloud provider integrations is changing.
+This means that users will need to confirm their Kubernetes configurations,
+and in some cases run external cloud controller managers. These changes are
+taking effect in Kubernetes version 1.29, read on to learn if you are affected
+and what changes you will need to make.
+
+These updated default settings affect a large proportion of Kubernetes users,
+and **will require changes** for users who were previously using the in-tree
+provider integrations. The legacy integrations offered compatibility with
+Azure, AWS, GCE, OpenStack, and vSphere; however for AWS and OpenStack the
+compiled-in integrations were removed in Kubernetes versions 1.26 and 1.27,
+respectively.
+
+## What has changed?
+
+At the most basic level, two [feature gates][fg] are changing their default
+value from false to true. Those feature gates, `DisableCloudProviders` and
+`DisableKubeletCloudCredentialProviders`, control the way that the
+[kube-apiserver][kapi], [kube-controller-manager][kcm], and [kubelet][kubelet]
+invoke the cloud provider related code that is included in those components.
+With these feature gates defaulting to true, the former behavior that was
+provided through the `--cloud-provider` command line flag is changing to only
+recognize the `external` value.
+
+Let's see what the [official Kubernetes documentation][fg] says about these
+feature gates:
+
+> `DisableCloudProviders`: Disables any functionality in `kube-apiserver`,
+> `kube-controller-manager` and `kubelet` related to the `--cloud-provider`
+> component flag.
+
+> `DisableKubeletCloudCredentialProviders`: Disable the in-tree functionality
+> in kubelet to authenticate to a cloud provider container registry for image
+> pull credentials.
+
+## What do you need to do?
+
+If you are upgrading from Kubernetes 1.28+ and are not on Azure, GCE, or
+vSphere then you are in luck, there are no changes you will need to make. If
```

Comment:
```suggestion
vSphere then there are no changes you will need to make. If
```

I would remove this idiom as it may be confusing (and structurally it introduces a run-on sentence that would need a semicolon if it stays).